apps/emqx_dashboard/src/emqx_dashboard.erl

```diff
@@ -58,21 +65,66 @@ start_listeners(Listeners) ->
                         ]),
                         {[Name | OkAcc], ErrAcc};
                     {error, _Reason} ->
+                        %% NOTE
                         %% Don't record the reason because minirest already does(too much logs noise).
                         {OkAcc, [Name | ErrAcc]}
                 end
             end,
             {[], []},
             listeners(ensure_ssl_cert(Listeners))
         ),
+    ok = emqx_dashboard_dispatch:regenerate_dispatch(OkListeners),
     case ErrListeners of
         [] ->
-            optvar:set(emqx_dashboard_listeners_ready, OkListeners),
```

Comment:
This didn't seem working properly.
The function is parametrized with `Listeners`, but the variable is not.
So it looks like a channel into `emqx_dashboard_listener` module for passing listener names to regenerate dispatch for.

The problem is that its value was also used for regenerating dispatches after config updates.
So in the scenario:
* We disable some listener (e.g. `http`)
* We enable it back, so it is passed to `start_listeners`.
* `emqx_dashboard_listeners_ready` get set to `['http:dashboard']`
* in subsequent config updates, only `http` gets regenerated.

We completely get rid of this complex and circle coupling of `emqx_dashboard` and `emqx_dashboard_listener`